### DISCUSSION TITLE
Errors in cpp project headers file

### DISCUSSION THREAD

**Discussion by nsmohan** [clangd/clangd#1893] [Q&A]
I have a cmake c++ project for which I was able to use the CMAKE_EXPORT_COMPILE_COMMANDS to generate a compile_commands.json file. When I open my source files everything works great. However when I open my header files they are filled with errors for undeclared variables and/or unknown types. How I have my project structed I do not have any includes in my header file. All my includes are in my source file. 

For Example: 

math.hpp

```
class Math
{
    int32_t add(a, b);
};

```

math.cpp

```
#include <cstdint>
#include "math.hpp"

int32_t Math::add(a, b)
{
    return a +b;
}

```

main.cpp

```
#include <iostream>
#include <cstdint>
#include "math.hpp"

int main()
{
    Math m;
    std::cout << m.add(4, 4) << std::endl;

    return 0;
}
```

In the above example If I open my header file clangd will complain about "int32_t" being a unknown type and ask me to include <cstdint> in math.hpp. 

How can I resolve this without including headers in my header file.

**Comment by HighCommander4** [comment]
The file `main.hpp` is not self-contained in this case. Clangd does not currently support files which are not self-contained, please see #45 for more discussion.
